### PR TITLE
feat: add bulk duplicate report action to Reports search page

### PR DESCRIPTION
**Pull request by Krishna2323**
### Fixed Issues
$ https://github.com/Expensify/App/issues/86683
PROPOSAL:

### Prerequisites: 

- This needs a full beta-access account

#### 1. Basic Bulk Report Duplication

1. Navigate to the **Reports** search page (Search > Reports tab)
2. Select 2-3 expense reports that you submitted (you are the submitter)
3. Click the bulk actions dropdown
4. Verify the **"Duplicate report"** option is visible
5. Select "Duplicate report"
6. Navigate to the LHN and verify a duplicate was created for each selected report
7. Verify each duplicate is named "Copy of [Original report name]"
8. Verify each duplicate contains the same cash expenses (merchant, amount, category, tag, currency) as its source
9. Verify each duplicated expense date is today's date

#### 2. Option Visibility — Only for Expense Reports You Submitted

1. Navigate to the Reports search page
2. Select an expense report where **you are not the submitter** (e.g. a report submitted to you for approval)
3. Verify the "Duplicate report" option is **not visible** in the bulk actions dropdown
4. Now select only reports you submitted
5. Verify the "Duplicate report" option **appears**

#### 3. Option Visibility — Only on Reports Tab

1. Navigate to the **Expenses** search page (Search > Expenses tab)
2. Select one or more expenses
3. Verify the "Duplicate report" option is **not visible** in the bulk actions dropdown
4. Switch to the **Reports** tab and select the same reports
5. Verify the "Duplicate report" option **is visible**

#### 4. Option Visibility — Requires Default Expense Policy

1. Leave all your workspaces so you have no default expense policy
2. Navigate to the Reports search page and select a report
3. Verify the "Duplicate report" option is **not visible**
4. Rejoin a workspace
5. Verify the "Duplicate report" option now **appears**

#### 5. Card and Scanning Expenses Excluded

1. Have a report with a mix of: cash expense, corporate card expense, and a scanning receipt
2. Select this report on the Reports search page and duplicate it
3. Open the duplicated report
4. Verify only the cash expense appears; card and scanning expenses are excluded

#### 6. Cross-Policy Duplication (Source Policy Accessible)

1. Have reports from two different workspaces where you are a member of both
2. Select both reports on the Reports search page and duplicate
3. Verify each duplicate is created under its **source workspace** (not your default)
4. Verify categories and tags from each source workspace are preserved

#### 7. Cross-Policy Duplication (Source Policy Inaccessible)

1. Leave a workspace, then navigate to the Reports search page
2. Select a report from the workspace you left
3. Duplicate it
4. Verify the duplicated report is created under your **default/primary workspace**
5. Verify the following transaction types are **skipped** (not included in the duplicate):
   - Per diem expenses
   - Distance expenses
   - Credit card imported expenses
   - Partially created or scanning expenses
   - Expenses with a "custom unit out of policy" violation
6. Verify that cash expenses are duplicated correctly

#### 8. Distance and Per Diem Expense Duplication

1. Have a report with distance and per diem expenses
2. Select it on the Reports search page and duplicate
3. Open the duplicated report
4. Verify the distance expense is present with correct distance and amount
5. Verify the per diem expense has correct subrates, dates, and amount

#### 9. Empty Report / No Eligible Transactions

1. Have a report where all transactions are card imports (no cash expenses)
2. Duplicate the report from the Reports search page
3. Verify an empty duplicated report is created with the name "Copy of [Original]"

#### 10. Selection Cleared After Duplication

1. Select multiple reports on the Reports search page
2. Duplicate them
3. Verify the selection is cleared after the action completes

- [x] Verify that no errors appear in the JS console

### Offline tests

1. Go offline (disconnect network)
2. Navigate to the Reports search page, select one or more reports, and duplicate
3. Verify the optimistic duplicate reports appear immediately in the LHN with correct names
4. Verify expenses show in the duplicated reports with correct amounts
5. Go back online
6. Verify the reports sync successfully with no duplicate reports or "not found" errors

### QA Steps
- Same as tests

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified there are no new alerts related to the `canBeMissing` param for `useOnyx`
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4510fc76bbf5df699a2575bfb49a276af90f3ed7/src/components/LocaleContextProvider.tsx#L80)
      - [x] If any non-english text was added/modified, I used [JaimeGPT](https://chatgpt.com/g/g-2dgOQl5VM-english-to-spanish-translator-aka-jaimegpt) to get English > Spanish translation. I then posted it in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4510fc76bbf5df699a2575bfb49a276af90f3ed7/src/components/LocaleContextProvider.tsx#L116-L123)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.ts or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If new assets were added or existing ones were modified, I verified that:
    - [x] The assets are optimized and compressed (for SVG files, run `npm run compress-svg`)
    - [x] The assets load correctly across all supported platforms.
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I added [unit tests](https://github.com/Expensify/App/blob/main/tests/README.md) for any new feature or bug fix in this PR to help automatically prevent regressions in this user flow.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/7863ff5b-9880-4fd9-95b5-963a35451882

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/dd373c2f-b623-458c-aac8-8965e862855d

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/fa0feb10-0d35-41c0-8471-06372ff59323

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/45f0e3bd-0c91-485f-ab18-b993562e3146

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/d9471dcd-203f-401c-8e47-821cc01d915a

https://github.com/user-attachments/assets/613f84d8-0b4e-445b-8cc2-d2aec06c644c

https://github.com/user-attachments/assets/ac30ba06-42f4-438f-85f1-7a534147e5f5

https://github.com/user-attachments/assets/e787e824-b4d9-473e-a68c-35c297a94187

https://github.com/user-attachments/assets/452c5c7f-2609-4677-b4f1-feaaf34737e9

https://github.com/user-attachments/assets/da319c40-f8a3-4842-9522-30b785770a02

https://github.com/user-attachments/assets/f9fdd73b-28fe-41ae-9e38-ea1be0c10c0e

https://github.com/user-attachments/assets/caae2edf-e8ec-4a75-88fb-a747c2c3c63d

https://github.com/user-attachments/assets/d807fc6f-9567-4d57-98c4-81234cb2135e

https://github.com/user-attachments/assets/0c6d12fd-06b3-442d-8b9c-f1988829d565

https://github.com/user-attachments/assets/cdd999a2-7a02-45d3-b8ce-2aa2b569cc6f

https://github.com/user-attachments/assets/2ca12f40-b7d7-4143-a75d-aa129119fab2

</details>

### CODE REVIEW

File: src/components/Search/BulkDuplicateReportHandler.tsx
Comment (hungvu193) [inline src/components/Search/BulkDuplicateReportHandler.tsx:1]: can we use it as a hook like this?  We can early return inside useEffect when duplicate report option is not visible.

```javascript
useBulkDuplicateReportHandler({selectedReports, allReports, onHandlerReady})
```

File: src/hooks/useBulkDuplicateReportAction.ts
Comment (hungvu193) [inline src/hooks/useBulkDuplicateReportAction.ts:1]: Please add unit test for this hook. It's mandatory to write unit test for every new hook that we added

File: src/hooks/useBulkDuplicateReportAction.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    const [recentWaypoints] = useOnyx(ONYXKEYS.NVP_RECENT_WAYPOINTS);
+    const [allTransactionViolations] = useOnyx(ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS);
+    const [allPolicies] = useOnyx(ONYXKEYS.COLLECTION.POLICY);
+    const [allPolicyCategories] = useOnyx(ONYXKEYS.COLLECTION.POLICY_CATEGORIES);
+    const [allPolicyTags] = useOnyx(ONYXKEYS.COLLECTION.POLICY_TAGS);
+
+    const handleDuplicateReports = () => {
+        const activePolicyExpenseChat = getPolicyExpenseChat(currentUserPersonalDetails.accountID, defaultExpensePolicy?.id);
+
+        const reportIDs = selectedReports.map((r) => r.reportID).filter((id): id is string => !!id);
```
Comment (hungvu193) [inline src/hooks/useBulkDuplicateReportAction.ts]: NAB:
```suggestion
        const reportIDs = selectedReports.map((r) => r.reportID).filter(Boolean);
```

File: src/hooks/useSearchBulkActions.ts [hunk 2]
```diff
@@ -861,6 +862,28 @@ function useSearchBulkActions({queryJSON}: UseSearchBulkActionsParams) {
...
+    const isDuplicateReportOptionVisible = useMemo(() => {
+        if (!isExpenseReportType || !defaultExpensePolicy || selectedReports.length === 0) {
+            return false;
+        }
+
+        return selectedReports.every((report) => {
+            if (!report.reportID) {
+                return false;
+            }
+            const fullReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`];
```
Comment (hungvu193) [inline src/hooks/useSearchBulkActions.ts]: Can we use the check `isDuplicateReportAction` directly for `selectedReport` instead of getting the report from the `reports` collection? AFAIK, `selectedReport` already has `type` and `ownerAccountID`

**Comment by hungvu193** [comment]
@MelvinBot review this

File: src/hooks/useBulkDuplicateReportAction.ts
Comment (Krishna2323) [reply src/hooks/useBulkDuplicateReportAction.ts:1]: Test added.

File: src/hooks/useBulkDuplicateReportAction.ts (hunk 1 above)
Comment (Krishna2323) [reply src/hooks/useBulkDuplicateReportAction.ts]: `.filter(Boolean)` doesn't narrow the TypeScript type — the result stays `(string | undefined)[]` instead of `string[]`, which would cause a type error since `bulkDuplicateReports` expects `reportIDs: string[]`. The explicit type guard `.filter((id): id is string => !!id)` is needed to satisfy TypeScript without an `as` cast.

File: src/components/Search/BulkDuplicateReportHandler.tsx
Comment (Krishna2323) [reply src/components/Search/BulkDuplicateReportHandler.tsx:1]: The component pattern is intentional—it lets us mount it only when `isDuplicateReportOptionVisible` is true, so the 12+ `useOnyx` subscriptions inside `useBulkDuplicateReportAction` (policies, categories, tags, violations, etc.) are only active when needed.

If we convert this to a hook, those subscriptions would always run on every search page load, even on the Expenses tab. This follows the same pattern as `BulkDuplicateHandler` used for expense duplication.

We could add an early return in `useEffect` to skip setting the handler, but the subscriptions would still run regardless.

File: src/hooks/useSearchBulkActions.ts (hunk 2 above)
Comment (Krishna2323) [reply src/hooks/useSearchBulkActions.ts]: Updated.

**Comment by Krishna2323** [comment]
@codex review

**Comment by Krishna2323** [comment]
@MelvinBot review and look for regressions

**Comment by Krishna2323** [comment]
@Valforte please run the generateTranslations script.

File: src/libs/actions/IOU/Duplicate.ts [hunk 3]
```diff
@@ -1039,5 +1040,101 @@ function bulkDuplicateExpenses({
...
+        }
+        const list = transactionsByReportID.get(transaction.reportID) ?? [];
+        list.push(transaction);
+        transactionsByReportID.set(transaction.reportID, list);
+    }
+
+    for (const reportID of reportIDs) {
+        const report = allReports[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`];
+        if (!report) {
+            continue;
```
[the remark this answers is not in the document]
Comment (Krishna2323) [reply src/libs/actions/IOU/Duplicate.ts]: Fixed — `useBulkDuplicateReportAction` now pre-filters `reportIDs` against `allReports` and bails early (without clearing selection) if none are present.

**Comment by Krishna2323** [comment]
@codex review

**Comment by hungvu193** [comment]
Codewise LGTM, please check the comment from AI. I'll do some tests today

**Comment by trjExpensify** [comment]
@hungvu193 did you test?

**Review by trjExpensify — DISMISSED** [review]
Part of the duplicate report project in #migrate. 👍

**Comment by hungvu193** [comment]
> @hungvu193 did you test? 

Yes. I'll take a another round of test today

**Comment by hungvu193** [comment]
@Krishna2323 Duplicate report sometimes doesn't highlight the duplicated row, can you check? It seems working fine with duplicate expense.

https://github.com/user-attachments/assets/a8a9b7c7-6a19-440b-b6a9-4ba4f69515a3

File: src/libs/actions/IOU/Duplicate.ts [hunk 4]
```diff
@@ -1039,5 +1044,102 @@ function bulkDuplicateExpenses({
...
+        }
+
+        const reportPolicy = allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${report.policyID}`];
+        const isSourcePolicyValid = !!reportPolicy && isPolicyAccessible(reportPolicy, currentUserLogin);
+        const targetPolicy = isSourcePolicyValid ? reportPolicy : defaultExpensePolicy;
+        const chatReportID = report.chatReportID ?? report.parentReportID;
+        const chatReport = chatReportID ? allReports[`${ONYXKEYS.COLLECTION.REPORT}${chatReportID}`] : undefined;
+        const parentChatReport = isSourcePolicyValid && chatReport ? chatReport : activePolicyExpenseChat;
```
[the remark this answers is not in the document]
Comment (Krishna2323) [reply src/libs/actions/IOU/Duplicate.ts]: Good catch. Fixed in [461a5aa](https://github.com/Expensify/App/commit/461a5aac21a8ab8ceb45894e6c77ed060bce801e)

File: src/hooks/useBulkDuplicateReportAction.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    const [allTransactionViolations] = useOnyx(ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS);
+    const [allPolicies] = useOnyx(ONYXKEYS.COLLECTION.POLICY);
+    const [allPolicyCategories] = useOnyx(ONYXKEYS.COLLECTION.POLICY_CATEGORIES);
+    const [allPolicyTags] = useOnyx(ONYXKEYS.COLLECTION.POLICY_TAGS);
+
+    const handleDuplicateReports = () => {
+        const activePolicyExpenseChat = getPolicyExpenseChat(currentUserPersonalDetails.accountID, defaultExpensePolicy?.id);
+
+        const reportsMap = allReports ?? {};
+        const reportIDs = selectedReports.map((r) => r.reportID).filter((id): id is string => !!id && !!reportsMap[`${ONYXKEYS.COLLECTION.REPORT}${id}`]);
```
[the remark this answers is not in the document]
Comment (Krishna2323) [reply src/hooks/useBulkDuplicateReportAction.ts]: The filter is intentional — `bulkDuplicateReports` requires the full report object from `allReports` to perform duplication (it needs `policyID`, `chatReportID`, `reportName`, transaction data, etc. that aren't available in the partial `selectedReports` data). The benefit of pre-filtering is that we preserve the user's selection instead of clearing it on a no-op. In practice, `OpenApp` hydrates `COLLECTION.REPORT` with all accessible reports at app startup, so selected reports should always be present by the time the user reaches the search page.

@MelvinBot can you please also confirm?

**Comment by Valforte** [comment]
Running translations

**Comment by Krishna2323** [comment]
> Duplicate report sometimes doesn't highlight the duplicated row, can you check? It seems working fine with duplicate expense.



@hungvu193 This seems to be an existing issue, it also doesn't highlight for new expense creation:



https://github.com/user-attachments/assets/3e8ed944-c73d-4950-a5b5-fa0536991892

**Comment by hungvu193** [comment]
> This seems to be an existing issue, it also doesn't highlight for new expense creation:

@Krishna2323 Sounds good. Can you resolve the linting in the meantime?

**Comment by Krishna2323** [comment]
@codex review

**Comment by trjExpensify** [comment]
@Krishna2323 where are you at with these failing jest tests?

File: src/libs/actions/IOU/Duplicate.ts [hunk 6]
```diff
@@ -1049,5 +1059,125 @@ function bulkDuplicateExpenses({
...
+    for (const selectedReport of selectedReportsParam) {
+        const reportID = selectedReport.reportID;
+        if (!reportID) {
+            continue;
+        }
+
+        const report =
+            (searchData?.[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`] as OnyxTypes.Report | undefined) ??
+            allReports[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`] ??
+            (selectedReport as OnyxTypes.Report);
```
[the remark this answers is not in the document]
Comment (Krishna2323) [reply src/libs/actions/IOU/Duplicate.ts]: Fixed. Added a guard that skips reports only available via the `selectedReport` fallback — when a `reportID` isn't found in `searchData` or `allReports` AND has no transactions, it's treated as a stale selection and skipped. Legitimate empty reports (present in `searchData` or `allReports`) are still duplicated normally, since duplicating empty reports is an intentional feature (`isDuplicateReportAction` allows it).

**Comment by Krishna2323** [comment]
@hungvu193 @Valforte I've added a beta gate for this feature because bulk features typically have many edge cases. I've worked on three bulk features, and both of the previous ones had beta gates to avoid reverts. WDYT? Let me know if you have a different opinion. IMO, bulk features are harder to test thoroughly since they involve cross-policy interactions.

**Comment by Krishna2323** [comment]
@codex review

**Comment by hungvu193** [comment]
> I've added a beta gate for this feature because bulk features typically have many edge cases. I've worked on three bulk features, and both of the previous ones had beta gates to avoid reverts.

Yep. I agree with this one 👍

**Comment by hungvu193** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified that the composer does not automatically focus or open the keyboard on mobile unless explicitly intended. This includes checking that returning the app from the background does not unexpectedly open the keyboard.
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: HybridApp
    - [x] Android: mWeb Chrome
    - [x] iOS: HybridApp
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4510fc76bbf5df699a2575bfb49a276af90f3ed7/src/components/LocaleContextProvider.tsx#L80)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4510fc76bbf5df699a2575bfb49a276af90f3ed7/src/components/LocaleContextProvider.tsx#L116-L123)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.ts or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] For any bug fix or new feature in this PR, I verified that sufficient [unit tests](https://github.com/Expensify/App/blob/main/tests/README.md) are included to prevent regressions in this flow.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: HybridApp</summary>



https://github.com/user-attachments/assets/4fc2d31f-00f9-4ac8-8e05-889b32544e05

</details>

<details>
<summary>Android: mWeb Chrome</summary>



https://github.com/user-attachments/assets/7f8a5021-708a-40a3-a906-4668f7a6bb6a

</details>

<details>
<summary>iOS: HybridApp</summary>




https://github.com/user-attachments/assets/8a99f02b-a41e-48ac-99bf-a352d2708ab6

</details>

<details>
<summary>iOS: mWeb Safari</summary>



https://github.com/user-attachments/assets/07d19fd6-50fb-409e-be08-3f93351d02b8

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>




https://github.com/user-attachments/assets/cfa961b7-a078-4877-bc88-62f5b4e1102d


https://github.com/user-attachments/assets/82b7e6e6-d4e2-4c83-a6db-4941f0cd5e7a


</details>

**Review by hungvu193 — APPROVED** [review]

**Review by Valforte — APPROVED** [review]
Agree with the beta flag, LGTM!

**Comment by lanitochka17** [comment]
Deploy Blocker #https://github.com/Expensify/App/issues/88665 was identified to be related to this PR.

File: src/libs/actions/IOU/Duplicate.ts [hunk 7]
```diff
@@ -1049,5 +1055,128 @@ function bulkDuplicateExpenses({
...
+    for (const transaction of allTransactionSources) {
+        if (!transaction || !transaction.reportID || transaction.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE) {
+            continue;
+        }
+        const list = transactionsByReportID.get(transaction.reportID) ?? [];
+        list.push(transaction);
+        transactionsByReportID.set(transaction.reportID, list);
+    }
+
+    for (const selectedReport of selectedReportsParam) {
```
Comment (nkdengineer) [inline src/libs/actions/IOU/Duplicate.ts:1127]: When duplicating multiple reports with multiple transactions in each report, it blocks the UI because many APIs need to be called. This causes this [issue](https://github.com/Expensify/App/issues/98970).
